luarules/gadgets/game_autocolors.lua

```diff
@@ -359,25 +346,25 @@ local function UpdateTeamColors()
         local g = Spring.GetTeamRulesParam(teamID, "AutoTeamColorGreen")/255
         local b = Spring.GetTeamRulesParam(teamID, "AutoTeamColorBlue")/255
         
-        if iconDevModeEnabled then
-            spSetTeamColor(teamID, IconDevModeColor[1]/255, IconDevModeColor[2]/255, IconDevModeColor[3]/255)
-        elseif AnonymousModeEnabled then
+        if iconDevModeColor then
```

Comment:
```suggestion
        if iconDevModeEnabledModoption ~= 'disabled' then
```
Checking for the value `disabled` explicitly shows intent. Note that `disabled` is the internal key name, so this won't cause translation problems.

Reply:
What if the client send some trash? I specifically tried to check if the key exists in a table. Disabled or empty string, why not just ignore everything else?

Reply:
If the client is sending trash for modoption values, half the game is going to break. But it can left as-is if desired.

Reply:
One day they update the lobby and start to send "Disabled" with a capital D, this statement will lead us to set the colors based on 'nil':
` if iconDevModeEnabledModoption ~= 'disabled' then`